lib/Interpreter/Transaction.cpp

```diff
@@ -62,12 +62,22 @@ namespace cling {
   }
 
   NamedDecl* Transaction::containsNamedDecl(llvm::StringRef name) const {
-    for (auto I = decls_begin(), E = decls_end(); I < E; ++I)
+    for (auto I = decls_begin(), E = decls_end(); I < E; ++I) {
       for (auto DI : I->m_DGR) {
-        if (NamedDecl* ND = dyn_cast<NamedDecl>(DI))
+        if (NamedDecl* ND = dyn_cast<NamedDecl>(DI)) {
           if (name.equals(ND->getNameAsString()))
             return ND;
+        }
+        else if (LinkageSpecDecl* LSD = dyn_cast<LinkageSpecDecl>(DI)) {
```

Comment:
That single use is a use not a misuse: misuse can only happen if everyone else it using it differently.  So I don't understand that point if it was meant to be one. Instead, back to fixing the name: `findTopLevelNamedDecl`? I don't want two searches (`containsDecl()` and Sema lookup) where one is enough.